src/core/induction_variables/src/LoopGoverningIVUtility.cpp

```diff
@@ -243,12 +254,13 @@ void LoopGoverningIVUtility::updateConditionToCheckIfWeHavePastExitValue(
 
   /*
    * Check if the loop is a while one and the value used to compare against the
-   * exit condition value is not the PHI of the loop governing IV.
+   * exit condition value is not an instruction of the PHI of the loop governing
+   * IV.
    */
   auto &IV = this->attribution.getInductionVariable();
   if (this->isWhile
-      && (this->attribution.getValueToCompareAgainstExitConditionValue()
-          != IV.getLoopEntryPHI())) {
+      && (!IV.getAllInstructions().count(
```

Comment:
Can you specify the condition more inline with the return value of "count" ?
count returns an integer, so something like ...count(..) == 0